File: src/main/java/com/test/taskmanager/app/view/ListProcessScreen.java
package com.test.taskmanager.app.view;

import static com.test.taskmanager.domain.SortingType.CREATION_TIME;
import static com.test.taskmanager.domain.SortingType.ID;
import static com.test.taskmanager.domain.SortingType.PRIORITY;

import de.codeshelf.consoleui.prompt.ListResult;
import de.codeshelf.consoleui.prompt.PromtResultItemIF;
import io.vavr.control.Try;
import java.util.Map;
import lombok.Value;
import lombok.experimental.SuperBuilder;

@Value
@SuperBuilder
public class ListProcessScreen extends AbstractScreen<Void> {

    private static final String CREATION_TIME_ACTION = "creationTime";
    private static final String PRIORITY_ACTION = "priority";
    private static final String ID_ACTION = "id";

    @Override
    public Void invoke() {
        promptBuilder.createListPrompt()
            .name(ACTION)
            .message("Process list sort type:")
            .newItem(CREATION_TIME_ACTION).text("1 - Creation time").add()
            .newItem(PRIORITY_ACTION).text("2 - Priority").add()
            .newItem(ID_ACTION).text("3 - Id").add()
            .newItem(BACK_ACTION).text("4 - Back").add()
            .addPrompt();

        Map<String, ? extends PromtResultItemIF> result = Try.of(() -> prompt.prompt(promptBuilder.build()))
            .get();

        ListResult action = (ListResult) result.get(ACTION);

        switch (action.getSelectedId()) {
            case CREATION_TIME_ACTION:
                taskManager.listAll(CREATION_TIME)
                    .forEach(System.out::println);
                break;
            case PRIORITY_ACTION:
                taskManager.listAll(PRIORITY)
                    .forEach(System.out::println);
                break;
            case ID_ACTION:
                taskManager.listAll(ID)
                    .forEach(System.out::println);
                break;
            case BACK_ACTION:
                MainScreen.builder()
                    .taskManager(taskManager)
                    .build()
                    .invoke();
                break;
        }

        ListProcessScreen.builder()
            .taskManager(taskManager)
            .build()
            .invoke();

        return null;
    }
}

File: src/main/kotlin/com/test/taskmanager/app/view/ListProcessScreen.kt
package com.test.taskmanager.app.view

import com.test.taskmanager.domain.SortingType.*
import com.test.taskmanager.service.TaskManager
import de.codeshelf.consoleui.prompt.ListResult
import de.codeshelf.consoleui.prompt.PromtResultItemIF
import io.vavr.control.Try
import java.util.function.Consumer

class ListProcessScreen(taskManager: TaskManager) : AbstractScreen<Void?>(taskManager) {
    override fun invoke(): Void? {
        promptBuilder.createListPrompt()
                .name(ACTION)
                .message("Process list sort type:")
                .newItem(CREATION_TIME_ACTION).text("1 - Creation time").add()
                .newItem(PRIORITY_ACTION).text("2 - Priority").add()
                .newItem(ID_ACTION).text("3 - Id").add()
                .newItem(BACK_ACTION).text("4 - Back").add()
                .addPrompt()
        val result: Map<String, PromtResultItemIF> = Try.of { prompt.prompt(promptBuilder.build()) }
                .get()

        val action = result[ACTION] as ListResult

        when (action.selectedId) {
            CREATION_TIME_ACTION -> taskManager.listAll(CREATION_TIME)
                    .forEach(Consumer(::println))
            PRIORITY_ACTION -> taskManager.listAll(PRIORITY)
                    .forEach(Consumer(::println))
            ID_ACTION -> taskManager.listAll(ID)
                    .forEach(Consumer(::println))
            BACK_ACTION -> MainScreen(taskManager).invoke()
        }

        ListProcessScreen(taskManager).invoke()
        return null
    }

    companion object {
        private const val CREATION_TIME_ACTION = "creationTime"
        private const val PRIORITY_ACTION = "priority"
        private const val ID_ACTION = "id"
    }
}